File: Sample/ClientSource/samples/Lua/HelloLua/Classes/FrameManager.cpp
#include "DataRequest.h"
#include "FrameManager.h"
#include "CCScheduler.h"
//#include "UpdateEngine.h"
#include "CCDirector.h"
//using namespace NdUpdate;
using namespace ScutDataLogic;

CFrameManager::CFrameManager( void )
{
	cocos2d::CCDirector::sharedDirector()->getScheduler()->scheduleUpdateForTarget(this, 0, false);
}

CFrameManager::~CFrameManager( void )
{
	cocos2d::CCDirector::sharedDirector()->getScheduler()->unscheduleUpdateForTarget(this);
}

void CFrameManager::update( float dt )
{
	//执行网络数据处理
	CDataRequest::Instance()->PeekLUAData();
	//执行插件相关处理
	//CUpdateEngine::getInstance()->tick(dt);	
}

CFrameManager* CFrameManager::Instance()
{
	static CFrameManager g_Instance;
	return &g_Instance;
}

File: Sample/GameRanking/Client/release/lua/lib/FrameManager.lua
FrameManager = {}

function FrameManager:new(o)
    o = o or {}
    setmetatable(o, self)
    self.__index = self
    return o
end

function FrameManager:init()
    CCDirector:sharedDirector():getScheduler():scheduleScriptFunc(function()
        self:update()
    end, 0, false)
end

function FrameManager:update()
    ScutDataLogic.CDataRequest:Instance():PeekLUAData()
end